homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +75,46 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+@TYPES.register('BinarySensor')
+class BinarySensor(HomeAccessory):
+    """Generate a BinarySensor accessory as binary sensor."""
+
+    def __init__(self, hass, state, **kwargs):
+        """Initialize a BinarySensor accessory object."""
+        entity_id = state.entity_id
+        super().__init__(state.name, entity_id, CATEGORY_SENSOR, **kwargs)
+
+        self.hass = hass
+        self.entity_id = entity_id
+
+        service_map = {
+            'gas': ('CarbonMonoxideSensor', 'CarbonMonoxideDetected'),
+            'co2': ('CarbonDioxideSensor', 'CarbonDioxideDetected'),
+            'occupancy': ('OccupancySensor', 'OccupancyDetected'),
+            'opening': ('ContactSensor', 'ContactSensorState'),
+            'motion': ('MotionSensor', 'MotionDetected'),
+            'moisture': ('LeakSensor', 'LeakDetected'),
+            'smoke': ('SmokeSensor', 'SmokeDetected')}
+
+        device_class_key = 'homekit_device_class' \
+            if ('homekit_device_class' in state.attributes) \
+            else 'device_class'
+        device_class = state.attributes.get(device_class_key)
```

Comment:
Use `ATTR_DEVICE_CLASS` imported from `homeassistant.const` as key

Reply:
Just follow it